_posts/en/newsletters/2024-05-24-newsletter.md

```diff
@@ -0,0 +1,243 @@
+
+  - *Upgrade on re-establish:* also described in a [draft
+    specification][bolts #868], this allows changing many channel
+    parameters any time two nodes re-establish a data connection.
+    It doesn't deal directly with funding and commitment transaction
+    upgrades.
+
+  With all the options presented, Kirk-Cohen compares them in a table
+  listing their onchain costs, upsides, and downsides; she also compares
+  them against the onchain costs of not making any upgrades.  She draws
+  several conclusions, including: "I think that it makes sense to start
+  work on [both] parameter [and] commitment upgrades via [dynamic
+  commitments][bolts #1117], independently of how we choose to go about
+  upgrading to taproot channels. This gives us the ability to upgrade to
+  `option_zero_fee_htlc_tx` anchor channels, and provides a commitment
+  format upgrade mechanism that can be used to get us to V3 channels
+  (once specified)."
+
+- **Challenges in rewarding pool miners:** Ethan Tuttle [posted][tuttle
+  poolcash] to Delving Bitcoin to suggest that [mining pools][topic
+  pooled mining] could reward miners with [ecash][topic ecash] tokens
+  proportionate to the number of shares they mined.  The miners could
+  then immediately sell or transfer the tokens, or they could wait for
+  the pool to mine a block, at which point the pool would exchange the
+  tokens for satoshis.
+
+  Both criticisms and suggestions of the idea were posted.  We found
+  especially insightful a [reply][corallo pooldelay] by Matt Corallo
+  where he describes an underlying problem: there are no standardized
+  payment methods implemented by large pools that allow pool miners to
+  calculate their own payouts over short intervals.  Two commonly used
+  payout types are:
+
+  - *Pay per share (PPS):* this pays a miner proportionately to amount
+    of work they contributed even if a block isn't found.  Calculating
+    the proportionate payout for the block subsidy is easy, but
+    calculating it for transaction fees is more complicated.  Corallo
+    notes that most pools appear to be using an average of the fees
+    collected during the day the share was created, meaning the amount
+    paid per share can't be calculated until a day after the share was
+    mined.  Additionally, many pools may tweak the fee averages in a way
+    that varies by pool.
+
+  - *Pay per last _n_ shares (PPLNS):* rewards miners for shares found
+    near the time the pool finds a block.  However, a miner can only be
+    sure that a pool found a block if the miner themselves found it.
+    There's no way (in the short term) for a typical miner to know the
+    pool is providing them a correct payout.
+
+  This lack of information means miners don't have the ability to
+  quickly switch to a different pool if their main pool stops paying
+  them.  [Stratum v2][topic pooled mining] doesn't fix this, although
+  forthright pools can use a standardized message to tell miners that
```

Comment:
"Forthright" might be an unnecessarily complicated word here. It's not even used in the post itself.

![forthright](https://github.com/bitcoinops/bitcoinops.github.io/assets/43024885/185c5532-c360-4afb-97ea-4266cbd9a6bb)

```suggestion
  honest pools can use a standardized message to tell miners that
```